In [1]:
import pandas as pd
import numpy as np

In [2]:
dataset = pd.read_excel("Sample - Superstore (1).xls", sheet_name='Orders')
pd.set_option('display.max_columns', None)

In [3]:
dataset.head()

Row ID        Order ID Order Date  Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
1       2  CA-2016-152156 2016-11-08 2016-11-11    Second Class    CG-12520   
2       3  CA-2016-138688 2016-06-12 2016-06-16    Second Class    DV-13045   
3       4  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   
4       5  US-2015-108966 2015-10-11 2015-10-18  Standard Class    SO-20335   

     Customer Name    Segment        Country             City       State  \
0      Claire Gute   Consumer  United States        Henderson    Kentucky   
1      Claire Gute   Consumer  United States        Henderson    Kentucky   
2  Darrin Van Huff  Corporate  United States      Los Angeles  California   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale     Florida   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale     Florida   

   Postal Code Region       Product ID         Category Sub-Category  \
0        42420  South  FUR-BO-10001798        Furniture    Bookcases   
1        42420  South  FUR-CH-10000454        Furniture       Chairs   
2        90036   West  OFF-LA-10000240  Office Supplies       Labels   
3        33311  South  FUR-TA-10000577        Furniture       Tables   
4        33311  South  OFF-ST-10000760  Office Supplies      Storage   

                                        Product Name     Sales  Quantity  \
0                  Bush Somerset Collection Bookcase  261.9600         2   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400         3   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200         2   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775         5   
4                     Eldon Fold 'N Roll Cart System   22.3680         2   

   Discount    Profit  
0      0.00   41.9136  
1      0.00  219.5820  
2      0.00    6.8714  
3      0.45 -383.0310  
4      0.20    2.5164

In [4]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Row ID         9994 non-null   int64         
 1   Order ID       9994 non-null   object        
 2   Order Date     9994 non-null   datetime64[ns]
 3   Ship Date      9994 non-null   datetime64[ns]
 4   Ship Mode      9994 non-null   object        
 5   Customer ID    9994 non-null   object        
 6   Customer Name  9994 non-null   object        
 7   Segment        9994 non-null   object        
 8   Country        9994 non-null   object        
 9   City           9994 non-null   object        
 10  State          9994 non-null   object        
 11  Postal Code    9994 non-null   int64         
 12  Region         9994 non-null   object        
 13  Product ID     9994 non-null   object        
 14  Category       9994 non-null   object        
 15  Sub-Category   9994 n

In [5]:
dataset.corr()

Row ID  Postal Code     Sales  Quantity  Discount    Profit
Row ID       1.000000     0.009671 -0.001359 -0.004016  0.013480  0.012497
Postal Code  0.009671     1.000000 -0.023854  0.012761  0.058443 -0.029961
Sales       -0.001359    -0.023854  1.000000  0.200795 -0.028190  0.479064
Quantity    -0.004016     0.012761  0.200795  1.000000  0.008623  0.066253
Discount     0.013480     0.058443 -0.028190  0.008623  1.000000 -0.219487
Profit       0.012497    -0.029961  0.479064  0.066253 -0.219487  1.000000

In [6]:
dataset.describe()

Row ID   Postal Code         Sales     Quantity     Discount  \
count  9994.000000   9994.000000   9994.000000  9994.000000  9994.000000   
mean   4997.500000  55190.379428    229.858001     3.789574     0.156203   
std    2885.163629  32063.693350    623.245101     2.225110     0.206452   
min       1.000000   1040.000000      0.444000     1.000000     0.000000   
25%    2499.250000  23223.000000     17.280000     2.000000     0.000000   
50%    4997.500000  56430.500000     54.490000     3.000000     0.200000   
75%    7495.750000  90008.000000    209.940000     5.000000     0.200000   
max    9994.000000  99301.000000  22638.480000    14.000000     0.800000   

            Profit  
count  9994.000000  
mean     28.656896  
std     234.260108  
min   -6599.978000  
25%       1.728750  
50%       8.666500  
75%      29.364000  
max    8399.976000

# TOP 5 MOST PROFIT IN DATASET

In [33]:
dataset.sort_values(by=['Product Name'], inplace=True, ascending=False)

In [8]:
dataset.head()

Row ID        Order ID Order Date  Ship Date       Ship Mode  \
6826    6827  CA-2016-118689 2016-10-02 2016-10-09  Standard Class   
8153    8154  CA-2017-140151 2017-03-23 2017-03-25     First Class   
4190    4191  CA-2017-166709 2017-11-17 2017-11-22  Standard Class   
9039    9040  CA-2016-117121 2016-12-17 2016-12-21  Standard Class   
4098    4099  CA-2014-116904 2014-09-23 2014-09-28  Standard Class   

     Customer ID  Customer Name    Segment        Country         City  \
6826    TC-20980   Tamara Chand  Corporate  United States    Lafayette   
8153    RB-19360   Raymond Buch   Consumer  United States      Seattle   
4190    HL-15040   Hunter Lopez   Consumer  United States       Newark   
9039    AB-10105  Adrian Barton   Consumer  United States      Detroit   
4098    SC-20095   Sanjit Chand   Consumer  United States  Minneapolis   

           State  Postal Code   Region       Product ID         Category  \
6826     Indiana        47905  Central  TEC-CO-10004722       Technology   
8153  Washington        98115     West  TEC-CO-10004722       Technology   
4190    Delaware        19711     East  TEC-CO-10004722       Technology   
9039    Michigan        48205  Central  OFF-BI-10000545  Office Supplies   
4098   Minnesota        55407  Central  OFF-BI-10001120  Office Supplies   

     Sub-Category                                      Product Name     Sales  \
6826      Copiers             Canon imageCLASS 2200 Advanced Copier  17499.95   
8153      Copiers             Canon imageCLASS 2200 Advanced Copier  13999.96   
4190      Copiers             Canon imageCLASS 2200 Advanced Copier  10499.97   
9039      Binders  GBC Ibimaster 500 Manual ProClick Binding System   9892.74   
4098      Binders              Ibico EPK-21 Electric Binding System   9449.95   

      Quantity  Discount     Profit  
6826         5       0.0  8399.9760  
8153         4       0.0  6719.9808  
4190         3       0.0  5039.9856  
9039        13       0.0  4946.3700  
4098         5       0.0  4630.4755

# "TOP" PROFIT : 8399.97 , Customer Name: Tamara Chand, Order Date:2016-10-02, Day: Sunday , State: Indiana

In [9]:
dataset.sort_values(by=['Profit'], inplace=True, ascending=False)
dataset.head(1)

Row ID        Order ID Order Date  Ship Date       Ship Mode  \
6826    6827  CA-2016-118689 2016-10-02 2016-10-09  Standard Class   

     Customer ID Customer Name    Segment        Country       City    State  \
6826    TC-20980  Tamara Chand  Corporate  United States  Lafayette  Indiana   

      Postal Code   Region       Product ID    Category Sub-Category  \
6826        47905  Central  TEC-CO-10004722  Technology      Copiers   

                               Product Name     Sales  Quantity  Discount  \
6826  Canon imageCLASS 2200 Advanced Copier  17499.95         5       0.0   

        Profit  
6826  8399.976

# TOP 5 MOST SALES IN DATASET

In [10]:
dataset.sort_values(by=['Sales'], inplace=True, ascending=False)

In [11]:
dataset.head()

Row ID        Order ID Order Date  Ship Date       Ship Mode  \
2697    2698  CA-2014-145317 2014-03-18 2014-03-23  Standard Class   
6826    6827  CA-2016-118689 2016-10-02 2016-10-09  Standard Class   
8153    8154  CA-2017-140151 2017-03-23 2017-03-25     First Class   
2623    2624  CA-2017-127180 2017-10-22 2017-10-24     First Class   
4190    4191  CA-2017-166709 2017-11-17 2017-11-22  Standard Class   

     Customer ID Customer Name      Segment        Country           City  \
2697    SM-20320   Sean Miller  Home Office  United States   Jacksonville   
6826    TC-20980  Tamara Chand    Corporate  United States      Lafayette   
8153    RB-19360  Raymond Buch     Consumer  United States        Seattle   
2623    TA-21385  Tom Ashbrook  Home Office  United States  New York City   
4190    HL-15040  Hunter Lopez     Consumer  United States         Newark   

           State  Postal Code   Region       Product ID    Category  \
2697     Florida        32216    South  TEC-MA-10002412  Technology   
6826     Indiana        47905  Central  TEC-CO-10004722  Technology   
8153  Washington        98115     West  TEC-CO-10004722  Technology   
2623    New York        10024     East  TEC-CO-10004722  Technology   
4190    Delaware        19711     East  TEC-CO-10004722  Technology   

     Sub-Category                                       Product Name  \
2697     Machines  Cisco TelePresence System EX90 Videoconferenci...   
6826      Copiers              Canon imageCLASS 2200 Advanced Copier   
8153      Copiers              Canon imageCLASS 2200 Advanced Copier   
2623      Copiers              Canon imageCLASS 2200 Advanced Copier   
4190      Copiers              Canon imageCLASS 2200 Advanced Copier   

          Sales  Quantity  Discount     Profit  
2697  22638.480         6       0.5 -1811.0784  
6826  17499.950         5       0.0  8399.9760  
8153  13999.960         4       0.0  6719.9808  
2623  11199.968         4       0.2  3919.9888  
4190  10499.970         3       0.0  5039.9856

# "TOP" SALES:22638.48, Customer Name:Sean Miller, Order Date:2014-03-18, Day:Sunday,  State:Florida

In [12]:
dataset.sort_values(by=['Sales'], inplace=True, ascending=False)
dataset.head(1)

Row ID        Order ID Order Date  Ship Date       Ship Mode  \
2697    2698  CA-2014-145317 2014-03-18 2014-03-23  Standard Class   

     Customer ID Customer Name      Segment        Country          City  \
2697    SM-20320   Sean Miller  Home Office  United States  Jacksonville   

        State  Postal Code Region       Product ID    Category Sub-Category  \
2697  Florida        32216  South  TEC-MA-10002412  Technology     Machines   

                                           Product Name     Sales  Quantity  \
2697  Cisco TelePresence System EX90 Videoconferenci...  22638.48         6   

      Discount     Profit  
2697       0.5 -1811.0784

# TOP 5 LEAST PROFIT IN DATASET

In [13]:
dataset.sort_values(by=['Profit'], inplace=True, ascending=True)

In [14]:
dataset.head()

Row ID        Order ID Order Date  Ship Date       Ship Mode  \
7772    7773  CA-2016-108196 2016-11-25 2016-12-02  Standard Class   
683      684  US-2017-168116 2017-11-04 2017-11-04        Same Day   
9774    9775  CA-2014-169019 2014-07-26 2014-07-30  Standard Class   
3011    3012  CA-2017-134845 2017-04-17 2017-04-23  Standard Class   
4991    4992  US-2017-122714 2017-12-07 2017-12-13  Standard Class   

     Customer ID   Customer Name      Segment        Country         City  \
7772    CS-12505   Cindy Stewart     Consumer  United States    Lancaster   
683     GT-14635  Grant Thornton    Corporate  United States   Burlington   
9774    LF-17185     Luke Foster     Consumer  United States  San Antonio   
3011    SR-20425  Sharelle Roach  Home Office  United States   Louisville   
4991    HG-14965   Henry Goldwyn    Corporate  United States      Chicago   

               State  Postal Code   Region       Product ID         Category  \
7772            Ohio        43130     East  TEC-MA-10000418       Technology   
683   North Carolina        27217    South  TEC-MA-10004125       Technology   
9774           Texas        78207  Central  OFF-BI-10004995  Office Supplies   
3011        Colorado        80027     West  TEC-MA-10000822       Technology   
4991        Illinois        60653  Central  OFF-BI-10001120  Office Supplies   

     Sub-Category                               Product Name     Sales  \
7772     Machines  Cubify CubeX 3D Printer Double Head Print  4499.985   
683      Machines  Cubify CubeX 3D Printer Triple Head Print  7999.980   
9774      Binders  GBC DocuBind P400 Electric Binding System  2177.584   
3011     Machines  Lexmark MX611dhe Monochrome Laser Printer  2549.985   
4991      Binders       Ibico EPK-21 Electric Binding System  1889.990   

      Quantity  Discount     Profit  
7772         5       0.7 -6599.9780  
683          4       0.5 -3839.9904  
9774         8       0.8 -3701.8928  
3011         5       0.7 -3399.9800  
4991         5       0.8 -2929.4845

# TOP 5 LEAST SALES IN DATASET

In [15]:
dataset.sort_values(by=['Sales'], inplace=True, ascending=True)
dataset.head()

Row ID        Order ID Order Date  Ship Date       Ship Mode  \
4101    4102  US-2017-102288 2017-06-19 2017-06-23  Standard Class   
9292    9293  CA-2017-124114 2017-03-02 2017-03-02        Same Day   
8658    8659  CA-2016-168361 2016-06-21 2016-06-25  Standard Class   
4711    4712  CA-2014-112403 2014-03-31 2014-03-31        Same Day   
2106    2107  US-2014-152723 2014-09-26 2014-09-26        Same Day   

     Customer ID     Customer Name    Segment        Country          City  \
4101    ZC-21910  Zuschuss Carroll   Consumer  United States       Houston   
9292    RS-19765    Roland Schwarz  Corporate  United States          Waco   
8658    KB-16600       Ken Brennan  Corporate  United States       Chicago   
4711    JO-15280     Jas O'Carroll   Consumer  United States  Philadelphia   
2106    HG-14965     Henry Goldwyn  Corporate  United States      Mesquite   

             State  Postal Code   Region       Product ID         Category  \
4101         Texas        77095  Central  OFF-AP-10002906  Office Supplies   
9292         Texas        76706  Central  OFF-BI-10004022  Office Supplies   
8658      Illinois        60623  Central  OFF-BI-10003727  Office Supplies   
4711  Pennsylvania        19120     East  OFF-BI-10003529  Office Supplies   
2106         Texas        75150  Central  OFF-BI-10003460  Office Supplies   

     Sub-Category                                       Product Name  Sales  \
4101   Appliances  Hoover Replacement Belt for Commercial Guardsm...  0.444   
9292      Binders           Acco Suede Grain Vinyl Round Ring Binder  0.556   
8658      Binders  Avery Durable Slant Ring Binders With Label Ho...  0.836   
4711      Binders                      Avery Round Ring Poly Binders  0.852   
2106      Binders                                  Acco 3-Hole Punch  0.876   

      Quantity  Discount  Profit  
4101         1       0.8 -1.1100  
9292         1       0.8 -0.9452  
8658         1       0.8 -1.3376  
4711         1       0.7 -0.5964  
2106         1       0.8 -1.4016

In [16]:
gro =dataset.groupby(["State"]).mean()

In [17]:
gro.sort_values(by=['Sales'], inplace=True, ascending=True)

# STATE WISE MIN SALES AVERAGE

In [18]:
gro.head()

Row ID   Postal Code       Sales  Quantity  Discount  \
State                                                                     
South Dakota  4071.500000  57227.500000  109.630000  3.500000  0.000000   
Kansas        5223.833333  66824.916667  121.429583  3.083333  0.000000   
New Mexico    4493.270270  87609.621622  129.284378  4.081081  0.059459   
North Dakota  5256.571429  58103.000000  131.415714  4.285714  0.000000   
Oregon        5266.096774  97355.846774  140.573790  4.024194  0.288710   

                 Profit  
State                    
South Dakota  32.902358  
Kansas        34.851813  
New Mexico    31.273408  
North Dakota  32.878529  
Oregon        -9.600569

# STATE WISE MAX SALES AVERAGE

In [34]:
gro.sort_values(by=['Product Name'], inplace=True, ascending=False)
gro.head()

KeyError: 'Product Name'

In [ ]:
gro.sort_values(by=['Sales'], inplace=True, ascending=False)
gro.head()

# "STATE" WISE MIN "PROFIT" AVERAGE

In [20]:
pro =dataset.groupby(["State"]).mean()
pro.sort_values(by=['Profit'], inplace=True, ascending=True)
pro.head()

Row ID   Postal Code       Sales  Quantity  Discount  \
State                                                                       
Ohio            5023.441365  43880.247335  166.861697  3.750533  0.324947   
Colorado        4379.131868  80295.758242  176.418231  3.807692  0.316484   
North Carolina  4896.293173  28089.895582  223.305880  3.947791  0.283534   
Tennessee       5836.836066  37653.409836  167.551219  3.721311  0.291257   
Pennsylvania    4716.393526  19063.994889  198.487077  3.667802  0.328620   

                   Profit  
State                      
Ohio           -36.186304  
Colorado       -35.867351  
North Carolina -30.083985  
Tennessee      -29.189583  
Pennsylvania   -26.507598

# "STATE" WISE MAX "PROFIT" AVERAGE

In [21]:
pro =dataset.groupby(["State"]).mean()
pro.sort_values(by=['Profit'], inplace=True, ascending=False)
pro.head()

Row ID   Postal Code       Sales  Quantity  Discount  \
State                                                                     
Vermont       8277.181818   5408.000000  811.760909  4.545455  0.000000   
Rhode Island  5453.303571   2908.535714  404.070643  3.553571  0.021429   
Indiana       4891.073826  46923.503356  359.431946  3.879195  0.000000   
Montana       4095.866667  59491.733333  372.623467  3.733333  0.066667   
Minnesota     3557.808989  55418.247191  335.541011  3.719101  0.000000   

                  Profit  
State                     
Vermont       204.088936  
Rhode Island  130.100523  
Indiana       123.375411  
Montana       122.221900  
Minnesota     121.608847

# WHICH CATEGORY SALES MOST ?____ANSWER : Office Supplies

In [22]:
dataset["Category"].value_counts()

Office Supplies    6026
Furniture          2121
Technology         1847
Name: Category, dtype: int64

# WHICH  SUB CATEGORY SALES MOST ?____ANSWER : Binders

In [23]:
dataset["Sub-Category"].value_counts()

Binders        1523
Paper          1370
Furnishings     957
Phones          889
Storage         846
Art             796
Accessories     775
Chairs          617
Appliances      466
Labels          364
Tables          319
Envelopes       254
Bookcases       228
Fasteners       217
Supplies        190
Machines        115
Copiers          68
Name: Sub-Category, dtype: int64

# MOST SALES CATEGORY?     ANSWER : Technology_AVERAGE = 452.70

# MOST PROFIT CATEGORY?    ANSWER : Technology_AVERAGE = 78.75

# LEAST SALES CATEGORY?ANSWER:Office Supplies_AVERAGE =119.32

# LEAST PROFIT CATEGORY?   ANSWER : Furniture_AVERAGE = 8.69

In [24]:
dataset.groupby(["Category"]).mean()

Row ID   Postal Code       Sales  Quantity  Discount  \
Category                                                                     
Furniture        5041.643564  55726.556341  349.834887  3.785007  0.173923   
Office Supplies  4980.175075  54890.951211  119.324101  3.801195  0.157285   
Technology       5003.331890  55551.572279  452.709276  3.756903  0.132323   

                    Profit  
Category                    
Furniture         8.699327  
Office Supplies  20.327050  
Technology       78.752002

# MOST SALES SUB CATEGORY? ANSWER : Copiers , AVERAGE" = 2198.94

# MOST PROFIT SUB CATEGORY?    ANSWER : Copiers,AVERAGE = 817.90

# LEAST SALES SUB CATEGORY?ANSWER:Fasteners,AVERAGE =13.93

# LEAST PROFIT SUB CATEGORY?   ANSWER : Tables,AVERAGE = 55.56

In [25]:
dataset.groupby(["Sub-Category"]).mean()

Row ID   Postal Code        Sales  Quantity  Discount  \
Sub-Category                                                               
Accessories   5068.490323  57378.624516   215.974604  3.840000  0.078452   
Appliances    5166.854077  54185.703863   230.755710  3.710300  0.166524   
Art           4924.760050  54434.243719    34.068834  3.768844  0.074874   
Binders       5012.432699  54908.994091   133.560560  3.922521  0.372292   
Bookcases     5072.521930  56015.521930   503.859633  3.807018  0.211140   
Chairs        5059.162075  56622.737439   532.332420  3.818476  0.170178   
Copiers       5230.308824  56962.897059  2198.941618  3.441176  0.161765   
Envelopes     4755.523622  52463.507874    64.867724  3.566929  0.080315   
Fasteners     4687.958525  57631.626728    13.936774  4.211982  0.082028   
Furnishings   5060.213166  54211.525601    95.825668  3.723093  0.138349   
Labels        4992.456044  53716.991758    34.303055  3.846154  0.068681   
Machines      4881.982609  55344.939130  1645.553313  3.826087  0.306087   
Paper         5026.343066  55692.862044    57.284092  3.779562  0.074891   
Phones        4944.865017  53877.587177   371.211534  3.699663  0.154556   
Storage       4877.835697  54667.517730   264.590553  3.732861  0.074704   
Supplies      5229.231579  55966.094737   245.650200  3.405263  0.076842   
Tables        4929.981191  58331.749216   648.794771  3.890282  0.261285   

                  Profit  
Sub-Category              
Accessories    54.111788  
Appliances     38.922758  
Art             8.200737  
Binders        19.843574  
Bookcases     -15.230509  
Chairs         43.095894  
Copiers       817.909190  
Envelopes      27.418019  
Fasteners       4.375660  
Furnishings    13.645918  
Labels         15.236962  
Machines       29.432669  
Paper          24.856620  
Phones         50.073938  
Storage        25.152277  
Supplies       -6.258418  
Tables        -55.565771

In [26]:
dataset["Segment"].value_counts()

Consumer       5191
Corporate      3020
Home Office    1783
Name: Segment, dtype: int64

# Home Office Segment has greater average sales and profit 

In [27]:
dataset.groupby(["Segment"]).mean()

Row ID   Postal Code       Sales  Quantity  Discount  \
Segment                                                                  
Consumer     4928.015604  55649.895781  223.733644  3.760547  0.158141   
Corporate    5134.017881  54482.228477  233.823300  3.843709  0.158228   
Home Office  4968.565339  55051.998317  240.972041  3.782389  0.147128   

                Profit  
Segment                 
Consumer     25.836873  
Corporate    30.456667  
Home Office  33.818664

# WHICH SHIPING MODE USED MOST?   ANSWER:Standard Class 

# WHICH SHIPING MODE LESS USED?    ANSWER:Same Day

In [28]:
dataset["Ship Mode"].value_counts()

Standard Class    5968
Second Class      1945
First Class       1538
Same Day           543
Name: Ship Mode, dtype: int64

# NAME OF THE SHIPPING MODE WHICH HAS GREATER SALES AVERAGE? ANSWER= First Class   "228.49"

# NAME OF THE SHIPPING MODE WHICH HAS LESS SALES AVERAGE? ANSWER= Second Class  "236.08"

# NAME OF THE SHIPPING MODE WHICH HAS GREATER PROFIT AVERAGE? ANSWER= First Class  "31.83"

# NAME OF THE SHIPPING MODE WHICH HAS LESS PROFIT AVERAGE? ANSWER= Standard Class  "27.49"

In [29]:
dataset.groupby(["Ship Mode"]).mean()

Row ID   Postal Code       Sales  Quantity  Discount  \
Ship Mode                                                                   
First Class     4875.510403  54765.611834  228.497024  3.701560  0.164610   
Same Day        5128.909761  57536.082873  236.396179  3.609576  0.152394   
Second Class    4936.759383  55626.009254  236.089239  3.816452  0.138895   
Standard Class  5036.776977  54944.447051  227.583067  3.819873  0.160023   

                   Profit  
Ship Mode                  
First Class     31.839948  
Same Day        29.266591  
Second Class    29.535545  
Standard Class  27.494770

# LEAST 5 PROFIT AVERAGE BY CITY

In [30]:
city =dataset.groupby(["City"]).mean()
city.sort_values(by=['Profit'], inplace=True, ascending=True)

In [31]:
city.head()

Row ID   Postal Code       Sales  Quantity  Discount  \
City                                                                    
Bethlehem   7282.000000  18018.000000  337.926800  3.600000  0.380000   
Champaign   3865.000000  61821.000000  151.960000  5.000000  0.600000   
Oswego      9480.000000  60543.000000  107.326000  4.333333  0.600000   
Round Rock  3718.285714  78664.000000  693.436114  3.285714  0.274286   
Lancaster   5599.173913  43059.130435  215.031826  3.717391  0.315217   

                Profit  
City                    
Bethlehem  -200.619160  
Champaign  -182.352000  
Oswego     -178.709200  
Round Rock -169.061614  
Lancaster  -157.371052

# TOP 5 PROFIT AVERAGE BY CITY

In [32]:
city =dataset.groupby(["City"]).mean()
city.sort_values(by=['Profit'], inplace=True, ascending=False)
city.head()

Row ID   Postal Code        Sales  Quantity  Discount  \
City                                                                       
Jamestown     5707.000000  14701.000000  2354.395000  4.500000       0.0   
Independence  1856.000000  64055.000000  1208.685000  3.000000       0.0   
Lafayette     5582.483871  61757.225806   807.619355  3.354839       0.0   
Minneapolis   3815.608696  55407.000000   733.501739  3.478261       0.0   
Appleton      5533.500000  54915.000000   835.655000  6.000000       0.0   

                  Profit  
City                      
Jamestown     642.885750  
Independence  487.831500  
Lafayette     323.173794  
Minneapolis   296.721070  
Appleton      277.383150